util.go

```diff
@@ -81,3 +85,21 @@ func createNonce(length int) string {
 
 	return string(nonce)
 }
+
+func setTLSContext(ctx context.Context, caBundle []byte) context.Context {
+	if len(caBundle) == 0 {
+		return ctx
+	}
+	rootCAs, _ := x509.SystemCertPool()
```

Comment:
Can we handle the error here?
How about logging a warning and continuing with a new CertPool?